### PR TITLE
feat: Apply generic UI to existing portlet - MEED-2653 - Meeds-io/MIPs#81

### PR DESCRIPTION
**Pull request by SaraBoutej**
This PR apply the new widget wrapper component for all gamifications widget

### CODE REVIEW

File: portlets/src/main/webapp/vue-app/myContributions/components/MyContributions.vue [hunk 1]
```diff
@@ -15,18 +15,18 @@ along with this program; if not, write to the Free Software Foundation,
 Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 -->
 <template>
-  <gamification-overview-widget :see-all-url="achievementsURL" :loading="loading">
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/myContributions/components/MyContributions.vue:18]: Why not simply changing `gamification-overview-widget` to reuse `widget-wrapper` ?

File: portlets/src/main/webapp/vue-app/engagement-center-actions/main.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ export function init(isAdministrator, isProgramManager) {
         document.dispatchEvent(new CustomEvent('hideTopBarLoading'));
         window.setInterval(() => this.now = Date.now(), 1000);
       },
-      template: `<engagement-center-rules id="${appId}" :is-administrator="${isAdministrator}" :is-program-manager="${isProgramManager}" />`,
+      template: `<engagement-center-rules id="${appId}" :is-administrator="${isAdministrator}" :is-program-manager="${isProgramManager}" class="card-border-radius overflow-hidden" />`,
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center-actions/main.js]: Add class on corresponding `v-app`

File: portlets/src/main/webapp/vue-app/engagement-center-programs/main.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ export function init(isAdministrator, isProgramManager) {
         document.dispatchEvent(new CustomEvent('hideTopBarLoading'));
         window.setInterval(() => this.now = Date.now(), 1000);
       },
-      template: `<engagement-center-programs id="${appId}" :is-administrator="${isAdministrator}" :is-program-manager="${isProgramManager}" />`,
+      template: `<engagement-center-programs id="${appId}" :is-administrator="${isAdministrator}" :is-program-manager="${isProgramManager}" class="card-border-radius overflow-hidden" />`,
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center-programs/main.js]: Add class on corresponding `v-app`

File: portlets/src/main/webapp/vue-app/programsOverview/components/ProgramOverviewItem.vue [hunk 4]
```diff
@@ -22,13 +22,13 @@
...
         width="48"
-        class="border-color rounded my-auto me-auto ms-4"
+        class="border-color rounded my-auto me-auto"
         tile>
         <v-img :src="program.avatarUrl" />
       </v-list-item-avatar>
     </template>
     <template #content>
-      <v-list-item class="ps-0">
+      <v-list-item class="ps-0 pe-0">
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/programsOverview/components/ProgramOverviewItem.vue]:
```suggestion
      <v-list-item class="px-0">
```

File: portlets/src/main/webapp/vue-app/myContributions/components/MyContributions.vue (hunk 1 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/myContributions/components/MyContributions.vue:18]: done

File: portlets/src/main/webapp/vue-app/engagement-center-actions/main.js (hunk 2 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center-actions/main.js]: done

File: portlets/src/main/webapp/vue-app/engagement-center-programs/main.js (hunk 3 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center-programs/main.js]: done

File: portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue [hunk 5]
```diff
@@ -16,34 +16,33 @@
...
+          <template v-if="remainingCount">
+            <gamification-overview-widget-empty-row
+              v-for="index in remainingCount"
+              :key="index"
+              class="flex-grow-1" />
           </template>
-        </gamification-overview-widget-row>
+        </div>
+        <div v-else-if="!loading">
+          <gamification-overview-widget-row class="my-auto mx-4">
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue]:
```suggestion
          <gamification-overview-widget-row v-else-if="!loading" class="my-auto mx-4">
```

File: portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue [hunk 6]
```diff
@@ -18,14 +18,11 @@
...
+    :title="title"
     height="auto"
-    min-width="auto"
-    extra-class="pa-0 justify-space-between">
-    <template #title>
-      <slot name="title"></slot>
-    </template>
-    <template #content>
+    min-width="auto">
+    <template #default>
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue]: When it's about default slot, no need to add it inside a `<template>`
```suggestion
```

File: portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue [hunk 7]
```diff
@@ -70,21 +67,17 @@
...
+    height="240">
+    <template #action>
+      <v-btn
+        icon
+        @click="hideEmptyWidget">
+        <v-icon>fa-times</v-icon>
+      </v-btn>
     </template>
-    <template #content>
+    <template #default>
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue]:
```suggestion
```

File: portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue (hunk 5 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue]: done

File: portlets/src/main/webapp/vue-app/programsOverview/components/ProgramOverviewItem.vue (hunk 4 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/programsOverview/components/ProgramOverviewItem.vue]: done

File: portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue (hunk 7 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue]: done

File: portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue [hunk 8]
```diff
@@ -16,27 +16,24 @@
...
-          :program="program"
-          class="flex-grow-1" />
-        <template v-if="remainingCount">
-          <gamification-overview-widget-empty-row
-            v-for="index in remainingCount"
-            :key="index"
+      :title="$t('gamification.overview.programsOverviewTitle')"
+      :action-url="programLink"
+      :loading="loading">
+      <template>
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue]:
```suggestion
```

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue (hunk 8 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/programsOverview/components/ProgramsOverview.vue]: done

File: portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue [hunk 9]
```diff
@@ -18,73 +18,64 @@
...
+    height="240">
+    <template #action>
+      <v-btn
+        icon
+        @click="hideEmptyWidget">
+        <v-icon>fa-times</v-icon>
+      </v-btn>
     </template>
-    <template #content>
+    <template>
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/rulesOverviewWidget/components/RulesOverviewWidget.vue]: Useless `<template>`
```suggestion
```

**Review by boubaker — APPROVED** [review]

**Pull request merged**